desktop_version/src/Editor.cpp

```diff
@@ -1799,1950 +1667,1760 @@ void editorrenderfixed(void)
+            graphics.backgrounddrawn = false;
+            break;
+        case 2:
+            //Go back to editor
+            music.playef(11);
+            game.returnmenu();
+            map.nexttowercolour();
+            break;
+        }
+        break;
+    case Menu::ed_font:
+    {
+        uint8_t idx_selected = font::font_idx_options[game.currentmenuoption];
+        cl.level_font_name = font::get_main_font_name(idx_selected);
+        if (idx_selected == loc::get_langmeta()->font_idx)
+        {
+            loc::new_level_font = "";
+        }
+        else
+        {
+            loc::new_level_font = cl.level_font_name;
+        }
+        font::set_level_font(cl.level_font_name.c_str());
+        music.playef(11);
+        game.returnmenu();
+        map.nexttowercolour();
+        game.savestatsandsettings_menu();
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+static void start_at_checkpoint(void)
+{
+    extern editorclass ed;
+
+    // Scan the room for a start point or a checkpoint, the start point taking priority
+    int testeditor = -1;
+    bool startpoint = false;
+
+    for (size_t i = 0; i < customentities.size(); i++)
+    {
+        if (customentities[i].t == 16 || customentities[i].t == 10)
+        {
+            int tx = customentities[i].x / 40;
+            int ty = customentities[i].y / 30;
+            if (tx == ed.levx && ty == ed.levy)
             {
-                if (game.press_action || game.press_left || game.press_right || game.press_map
-                        || up_pressed || down_pressed || key.isDown(27))
+                startpoint = customentities[i].t == 16;
+                if (startpoint)
                 {
-                    game.jumpheld = true;
+                    // Oh, there's a start point! Let's use this.
+                    testeditor = i;
+                    break;
                 }
-                if ((game.press_action || game.press_map) && !ed.hooklist.empty())
+                else if (testeditor == -1)
                 {
-                    game.mapheld=true;
-                    ed.scripthelppage=1;
-                    key.enabletextentry();
-                    key.keybuffer="";
-                    ed.sbscript=ed.hooklist[(ed.hooklist.size()-1)-ed.hookmenu];
-                    ed.loadhookineditor(ed.sbscript);
-
-                    ed.sby=ed.sb.size()-1;
-                    ed.pagey=0;
-                    while(ed.sby>=ed.lines_visible-5)
-                    {
-                        ed.pagey++;
-                        ed.sby--;
-                    }
-                    key.keybuffer=ed.sb[ed.pagey+ed.sby];
-                    ed.sbx = UTF8_total_codepoints(ed.sb[ed.pagey+ed.sby].c_str());
-                    music.playef(11);
+                    // This is the first thing we found, so let's use it (unless we find a start point later)
+                    testeditor = i;
                 }
             }
         }
```

Comment:
This for-loop can be rewritten to be more self-documenting with less indentation. Unfortunately, this is not a GitHub suggestion because GitHub doesn't support suggestions on deleted lines, and with the way the overall diff works out, there's a bunch of deleted lines in this part of the diff. And suggestions are very broken on intermediate commits.
```c++
        startpoint = customentities[i].t == 16;
        const bool is_startpoint_or_checkpoint = startpoint ||
            customentities[i].t == 10;
        if (!is_startpoint_or_checkpoint)
        {
            continue;
        }

        const int tx = customentities[i].x / 40;
        const int ty = customentities[i].y / 30;
        const bool in_room = tx == ed.levx && ty == ed.levy;
        if (!in_room)
        {
            continue;
        }

        if (startpoint)
        {
            // Oh, there's a start point! Let's use this.
            testeditor = i;
            break;
        }
        else if (testeditor == -1)
        {
            // This is the first thing we found, so let's use it (unless we find a start point later)
            testeditor = i;
        }
```